Code/Framework/AzFramework/AzFramework/DocumentPropertyEditor/AdapterBuilder.cpp

```diff
@@ -0,0 +1,143 @@
+/*
+ * Copyright (c) Contributors to the Open 3D Engine Project.
+ * For complete copyright and license terms please see the LICENSE at the root of this distribution.
+ *
+ * SPDX-License-Identifier: Apache-2.0 OR MIT
+ *
+ */
+
+#include <AzFramework/DocumentPropertyEditor/AdapterBuilder.h>
+#include <AzFramework/DocumentPropertyEditor/DocumentSchema.h>
+
+namespace AZ::DocumentPropertyEditor
+{
+    AdapterBuilder::AdapterBuilder(RoutingAdapter* routingAdapter)
+        : m_routingAdapter(routingAdapter)
+    {
+    }
+
+    void AdapterBuilder::Value(Dom::Value value)
+    {
+        CurrentNode().SetNodeValue(value);
+    }
+
+    void AdapterBuilder::Attribute(Name attribute, Dom::Value value)
+    {
+        CurrentNode()[attribute] = AZStd::move(value);
+    }
+
+    bool AdapterBuilder::IsError() const
+    {
+        return !m_error.empty();
+    }
+
+    const AZStd::string& AdapterBuilder::GetError() const
```

Comment:
This can return a string_view instead?

Reply:
In this case it's already a string so const ref makes more sense IMO, it'll cheaply promote to `string_view` if needed and be (very slightly) faster otherwise.